diff --git a/BUILD.gn b/BUILD.gn
index 72d9a23..9c0532e 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -26,7 +26,8 @@ import("gn/ios.gni")
 
 # Skia public API, generally provided by :skia.
 config("skia_public") {
-  include_dirs = [ "." ]
+  include_dirs = [ ".", "/usr/local/include" ]
+  lib_dirs = [ "/usr/local/lib" ]
 
   defines = [
     "SK_CODEC_DECODES_BMP",
diff --git a/gn/toolchain/BUILD.gn b/gn/toolchain/BUILD.gn
index 85858b5..8d6a0ae 100644
--- a/gn/toolchain/BUILD.gn
+++ b/gn/toolchain/BUILD.gn
@@ -315,7 +315,7 @@ template("gcc_like_toolchain") {
         rspfile = "{{output}}.rsp"
         rspfile_content = "{{inputs}}"
         rm_py = rebase_path("../rm.py")
-        command = "$shell python3 \"$rm_py\" \"{{output}}\" && $ar rcs {{output}} @$rspfile"
+        command = "$shell python3 \"$rm_py\" \"{{output}}\" && $ar rcs {{output}} `cat $rspfile`"
       }
 
       outputs =

